### PR TITLE
Added Azure Blob Storage File and Container Loader

### LINKED ISSUE #1805: Document loader for Azure Blob storage
Lots of customers is asking if langchain have a document loader like AWS S3 or GCS for Azure Blob Storage as well. As you know Microsoft is a big partner for OpenAI , so there is a real need to have native document loader for Azure Blob storage as well. We will be very happy to see this feature ASAP.

### PR DESCRIPTION
**Pull request by eavanvalkenburg**
Added support for document loaders for Azure Blob Storage using a connection string. Fixes #1805

### CODE REVIEW

File: langchain/document_loaders/azure_blob_storage_file.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            ) from exc
+
+        client = BlobClient.from_connection_string(
+            conn_str=self.conn_str, container_name=self.container, blob_name=self.blob
+        )
+
+        with tempfile.TemporaryDirectory() as temp_dir:
+            file_path = f"{temp_dir}/{self.container}/{self.blob}"
+            os.makedirs(os.path.dirname(file_path), exist_ok=True)
+            print(file_path)
```
Comment (iMicknl) [inline langchain/document_loaders/azure_blob_storage_file.py:37]:
```suggestion
```

**Review by hwchase17 — APPROVED** [review]
thank you - this is awesome!

File: langchain/document_loaders/azure_blob_storage_container.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+"""Loading logic for loading documents from an Azure Blob Storage file."""
```
Comment (iMicknl) [inline langchain/document_loaders/azure_blob_storage_container.py:1]:
```suggestion
"""Loading logic for loading documents from an Azure Blob Storage directory."""
```

**Review by hwchase17 — COMMENTED** [review]
is it possible to get an example notebook? i think thats the only thing missing

**Comment by eavanvalkenburg** [comment]
@hwchase17 I added sample notebooks.

**Pull request merged**